libethereum/Block.cpp

```diff
@@ -567,6 +567,18 @@ tuple< TransactionReceipts, unsigned > Block::syncEveryone( BlockChain const& _b
             LOG( m_loggerError ) << "FAILED transaction after consensus! " << ex.what();
         }
     }
+#ifdef MIRAGE
+    auto lastRewardedBlockNumber = m_state.getLastRewardedBlockNumber();
+    if ( lastRewardedBlockNumber < m_currentBlock.number() ) {
+        auto blockTimestamp = m_currentBlock.timestamp();
+        auto reward = _bc.sealEngine()->blockReward( blockTimestamp, m_currentBlock.number() );
```

Comment:
please use `m_previousBlock` for `reward` and `removeEmptyAccounts`. you can see how it is done in `State::execute`